server/src/main/java/org/apache/druid/metadata/SegmentsMetadataManager.java

```diff
@@ -125,6 +125,25 @@ Optional<Iterable<DataSegment>> iterateAllUsedNonOvershadowedSegmentsForDatasour
       boolean requiresLatest
   );
 
+  /**
+   * Returns an iterable to go over un-used segments for a given datasource over an optional interval.
+   * The order in which segments are iterated is from earliest start-time, with ties being broken with earliest end-time
+   * first. Note: the iteration may not be as trivially cheap as,
+   * for example, iteration over an ArrayList. Try (to some reasonable extent) to organize the code so that it
+   * iterates the returned iterable only once rather than several times.
+   *
+   * @param datasource the name of the datasource.
+   * @param interval   the interval to search over.
+   * @param limit      the maximum number of results to return.
+   * @param offset     the offset from which to search for results.
+   */
+  Iterable<DataSegment> iterateAllUnusedSegmentsForDatasource(
+      String datasource,
+      @Nullable Interval interval,
+      @Nullable Integer limit,
```

Comment:
I think it might be a good idea to go ahead and add a sort order now too

Reply:
Thanks! Good suggestion. Added